src/main/java/seedu/duke/command/EditCommand.java

```diff
@@ -2,36 +2,45 @@
 
 import seedu.duke.model.Item;
 import seedu.duke.model.ItemContainer;
+import seedu.duke.model.exception.IllegalArgumentException;
+import seedu.duke.model.exception.ItemNotExistException;
 
 /**
  * The command that edits a selected item.
  */
 public class EditCommand extends Command {
-    private final Item selectedItem;
-    private final Item updatedItem;
-    private static final String UPDATE_COMPLETE_MESSAGE = "This item has been updated."; //to be added to UI part later
-
+    private final String name;
+    private final String purcaseCost;
+    private final String sellingPrice;
+    private static final String UPDATE_COMPLETE_MESSAGE = "This item has been updated.";
+    //to be added to UI part later
 
     /**
      * The EditCommand constructor.
      *
-     * @param name the name of the selected item
+     * @param name         the name of the selected item
      * @param purchaseCost the new cost of the item
      * @param sellingPrice the new price of the item
-     * @param list the itemContainer where the selected item is stored in
      */
-    public EditCommand(String name, String purchaseCost, String sellingPrice, ItemContainer list) {
-        this.selectedItem = list.getItem(name);
-        this.updatedItem = new Item(name, purchaseCost, sellingPrice);
+    public EditCommand(String name, String purchaseCost, String sellingPrice) {
+        this.name = name;
+        this.purcaseCost = purchaseCost;
+        this.sellingPrice = sellingPrice;
     }
 
     /**
      * Executes the update operation.
      *
      * @param list the itemContainer to remove the item from
+     * @throws ItemNotExistException when cannot find any item with the name
+     * @throws NullPointerException when the name specified is null
+     * @throws IllegalArgumentException when the argument is invalid
```

Comment:
#35 I think the command should create its own exception classes, for example, an IllegalCommandException(), and take in a string to specify the reason for the error, then whoever is using this command package no need to import the model class to be able to capture the exception.